### PR TITLE
Thematize `Icon` component

### LINKED ISSUE #2208: Thematize `Icon` component
### Description

Yamada UI components are tied to a theme. However, the `Icon` component does not have a theme.

### Problem Statement/Justification

Themes have `defaultProps`, which can be utilized to share `props` throughout the project. However, the `Icon` component does not have a theme, so it cannot take advantage of them.

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by cidkumagai**
Closes #2208 

## Description

Thematize Icon component

### CODE REVIEW

File: .changeset/fifty-bulldogs-prove.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@yamada-ui/icon": patch
```
Comment (hirotomoyamada) [inline .changeset/fifty-bulldogs-prove.md:2]: Change `patch` to `minor`.

File: .changeset/fifty-bulldogs-prove.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+"@yamada-ui/theme": patch
```
Comment (hirotomoyamada) [inline .changeset/fifty-bulldogs-prove.md:3]: Change `patch` to `minor`.

File: packages/components/icon/src/icon.tsx [hunk 3]
```diff
@@ -21,22 +21,15 @@ export type IconProps = Omit<SVGAttributes<SVGElement>, keyof UIProps> &
  * @see Docs https://yamada-ui.com/components/media-and-icons/icon
  */
 export const Icon = forwardRef<IconProps, "svg">(
-  (
-    { as: element, viewBox, size: fontSize, className, __css, ...rest },
-    ref,
-  ) => {
+  ({ as: element, viewBox, size: fontSize, ...props }, ref) => {
```
Comment (hirotomoyamada) [inline packages/components/icon/src/icon.tsx:24]:
```
({ as: element, viewBox, size: fontSize, ...props }, ref) => {
```

to

```
(props, ref) => {
```

File: packages/components/icon/src/icon.tsx [hunk 4]
```diff
@@ -21,22 +21,15 @@ export type IconProps = Omit<SVGAttributes<SVGElement>, keyof UIProps> &
...
     )
 
     const css: CSSUIObject = {
-      w: "1em",
-      h: "1em",
-      display: "inline-block",
-      lineHeight: "1em",
-      flexShrink: 0,
-      color: "currentColor",
-      ...__css,
```
Comment (hirotomoyamada) [inline packages/components/icon/src/icon.tsx:39]: Do not delete `__css`.It is used by other components.

File: packages/components/icon/src/icon.tsx [hunk 5]
```diff
@@ -21,22 +21,15 @@ export type IconProps = Omit<SVGAttributes<SVGElement>, keyof UIProps> &
  * @see Docs https://yamada-ui.com/components/media-and-icons/icon
  */
 export const Icon = forwardRef<IconProps, "svg">(
-  (
-    { as: element, viewBox, size: fontSize, className, __css, ...rest },
-    ref,
-  ) => {
+  ({ as: element, viewBox, size: fontSize, ...props }, ref) => {
+    const [styles, { className, ...rest }] = useComponentStyle("Icon", props)
```
Comment (hirotomoyamada) [inline packages/components/icon/src/icon.tsx:25]:
```
const [styles, { className, ...rest }] = useComponentStyle("Icon", props)
```

to

```
const [styles, mergedProps] = useComponentStyle("Icon", props)
const {
  className,
  as: element,
  viewBox,
  size: fontSize,
  ...rest
} = omitThemeProps(mergedProps)
```

Use `omitThemeProps` to exclude `props` from the theme.

Comment (cidkumagai) [reply packages/components/icon/src/icon.tsx:25]: @hirotomoyamada 
```
型 'OmitProps<PropsOf<"svg">, "string" | "_hover" | "_nativeHover" | "_active" | "_nativeActive" | "_target" | "_notTarget" | "_nativeTarget" | "_focus" | "_nativeFocus" | ... 476 more ... | "size"> & ... 5 more ... & { ...; }' の引数を型 'ThemeProps' のパラメーターに割り当てることはできません。
  プロパティ 'size' の型に互換性がありません。
    型 'UIValue<number | (string & {}) | "small" | "-moz-initial" | "inherit" | "initial" | "revert" | "revert-layer" | "unset" | "large" | "medium" | "x-large" | "x-small" | "xx-large" | ... 17 more ... | "9xl"> | undefined' を型 'UIValue<string> | undefined' に割り当てることはできません。
      型 'number' を型 'UIValue<string> | undefined' に割り当てることはできません。
```

I get this error, can you tell me how to fix it?

Comment (hirotomoyamada) [reply packages/components/icon/src/icon.tsx:25]: Add `ThemeProps<"Icon">`.

Comment (cidkumagai) [reply packages/components/icon/src/icon.tsx:25]: Thank you very much!
Correction.

**Comment by cidkumagai** [comment]
@hirotomoyamada 
I have fixed it, please review.

**Comment by hirotomoyamada** [comment]
@cidkumagai 

Since the target branch is `main`, it is better to rebuild the PR once.

**Comment by cidkumagai** [comment]
@hirotomoyamada 
I understand. I'll remake it.

**Pull request closed without merging**